src/app/components/category-products/category-products.component.ts

```diff
@@ -1,25 +1,87 @@
-import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ViewEncapsulation,
+} from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { Observable, shareReplay, switchMap, tap } from 'rxjs';
+import {
+  combineLatest,
+  Observable,
+  filter,
+  map,
+  shareReplay,
+  switchMap,
+  tap,
+  of,
+} from 'rxjs';
+import { ProductWithStarsQueryModel } from 'src/app/query-models/product-with-stars.query-model';
 import { CategoryModel } from '../../models/category.model';
+import { ProductModel } from '../../models/product.model';
 import { CategoriesService } from '../../services/categories.service';
+import { ProductsService } from '../../services/products.service';
 
 @Component({
   selector: 'app-category-products',
   styleUrls: ['./category-products.component.scss'],
   templateUrl: './category-products.component.html',
   encapsulation: ViewEncapsulation.Emulated,
-  changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CategoryProductsComponent {
-  readonly categories$: Observable<CategoryModel[]> = this._categoriesService.getAllCategories().pipe(shareReplay(1));
-  readonly category$: Observable<CategoryModel[]|any> = this._activatedRoute.params.pipe(
-    switchMap((data) => {
-     return this._categoriesService.getOneCategory(data['categoryId'])}
-   ),tap(()=>{
-    console.log(this.category$)
-   })
-  )
-  constructor(private _categoriesService: CategoriesService, private _activatedRoute: ActivatedRoute) {
+  readonly categories$: Observable<CategoryModel[]> = this._categoriesService
+    .getAllCategories()
+    .pipe(shareReplay(1));
+  readonly category$: Observable<CategoryModel[] | any> =
+    this._activatedRoute.params.pipe(
+      switchMap((data) => {
```

Comment:
in my opinion, you should add shareReplay, because you have subscribe in html and in other variable